solidstart/github-oauth/src/routes/login/github/callback.ts

```diff
@@ -38,8 +31,9 @@ export async function GET(event: APIEvent) {
 
 		if (existingUser) {
 			const session = await lucia.createSession(existingUser.id, {});
-			appendHeader(event, "Set-Cookie", lucia.createSessionCookie(session.id).serialize());
-			return sendRedirect(event, "/");
+			const cookie = lucia.createSessionCookie(session.id);
+
+			setCookie(cookie.name, cookie.value, cookie.attributes);
```

Comment:
You should be returning here, otherwise you throw an error because you try to add the user to the database again